File: Problems/0001.Two-Sum/TwoSum.go
package leetcode

func twoSum(nums []int, target int) []int {
	rec := make(map[int]int)
	for idx, n := range nums {
		m := target - n
		if loc, isExist := rec[m]; isExist {
			return []int{loc, idx}
		}
		rec[n] = idx
	}
	return []int{}
}

File: leetcode/0001.Two-Sum/TwoSum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        rec = {}
        for idx, n in enumerate(nums):
            m = target - n
            if m in rec:
                return [rec[m], idx]
            rec[n] = idx
